tests/run-pass/floats.rs

```diff
@@ -1,37 +1,84 @@
 fn main() {
+    // basic arithmetic
     assert_eq!(6.0_f32*6.0_f32, 36.0_f32);
     assert_eq!(6.0_f64*6.0_f64, 36.0_f64);
     assert_eq!(-{5.0_f32}, -5.0_f32);
+    assert_eq!(-{5.0_f64}, -5.0_f64);
+    // infinities, NaN
     assert!((5.0_f32/0.0).is_infinite());
+    assert!((5.0_f64/0.0).is_infinite());
     assert!((-5.0_f32).sqrt().is_nan());
+    assert!((-5.0_f64).sqrt().is_nan());
+    // byte-level transmute
     let x: u64 = unsafe { std::mem::transmute(42.0_f64) };
     let y: f64 = unsafe { std::mem::transmute(x) };
     assert_eq!(y, 42.0_f64);
+    let x: u32 = unsafe { std::mem::transmute(42.0_f32) };
+    let y: f32 = unsafe { std::mem::transmute(x) };
+    assert_eq!(y, 42.0_f32);
 
+    // f32 casts
     assert_eq!(5.0f32 as u32, 5);
+    assert_eq!(-5.0f32 as u32, 0);
     assert_eq!(5.0f32 as i32, 5);
     assert_eq!(-5.0f32 as i32, -5);
+    assert_eq!(std::f32::MAX as i32, i32::MAX);
+    assert_eq!(std::f32::INFINITY as i32, i32::MAX);
+    assert_eq!(std::f32::MAX as u32, u32::MAX);
+    assert_eq!(std::f32::INFINITY as u32, u32::MAX);
+    assert_eq!(std::f32::MIN as i32, i32::MIN);
+    assert_eq!(std::f32::NEG_INFINITY as i32, i32::MIN);
+    assert_eq!(std::f32::MIN as u32, 0);
+    assert_eq!(std::f32::NEG_INFINITY as u32, 0);
+    assert_eq!(std::f32::NAN as i32, 0);
+    assert_eq!(std::f32::NAN as u32, 0);
+    assert_eq!(u128::MAX as f32, std::f32::INFINITY);
+    assert_eq!((u32::MAX-127) as f32 as u32, u32::MAX); // rounding loss
+    assert_eq!((u32::MAX-128) as f32 as u32, u32::MAX-255); // rounding loss
```

Comment:
@hannah-kruppe do these two (and the two "rounding loss" tests below) look like correct behavior to you? I experimentally determined these to be the closest to `u32::MAX` where the output of the roundtrip cast changes.